src/apps/content-editor/src/app/views/ItemList/ItemList.js

```diff
@@ -56,708 +63,629 @@ export default connect((state, props) => {
-          this.props.filters[this.props.modelZUID].status) ||
-        "all"
+})(function ItemList(props) {
+  const history = useHistory();
+  const _isMounted = useRef(false);
+
+  // table, list dimensions
+  const table = useRef();
+  const list = useRef();
+  const [height, setHeight] = useState(1000);
+  const [width, setWidth] = useState(1000);
+
+  // saving, loading states
+  const [saving, setSaving] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [loadingSecondary, setLoadingSecondary] = useState(false);
+
+  const [items, setItems] = useState([]);
+  const [itemCount, setItemCount] = useState(0);
+  const [fields, setFields] = useState([]);
+  const [filter, setFilter] = useState(
+    props.filters[props.modelZUID]
+      ? props.filters[props.modelZUID]
+      : DEFAULT_FILTER
+  );
+
+  const [shouldRunFilters, setShouldRunFilters] = useState(false);
+  const [shouldUpdateItemCount, setShouldUpdateItemCount] = useState(false);
+
+  // track isMounted
+  useEffect(() => {
+    _isMounted.current = true;
+    return () => {
+      _isMounted.current = false;
     };
-
-    componentDidMount() {
-      this._isMounted = true;
-
-      if (this.props.model) {
-        // Render rows we have available right away
-        this.updateRows();
-
-        // apply filters
-        if (this.props.filters[this.props.modelZUID]) {
-          this.runAllFilters();
-        }
-        // Hit API to check for new rows
-        this.load(this.props.modelZUID);
+  }, []);
+
+  // redirect to single entry
+  useEffect(() => {
+    if (
+      props.model &&
+      (props.model.name === "clippings" || props.model.type === "templateset")
+    ) {
+      // First item is always the column row
+      if (items.length === 2) {
+        // redirect to the single entry in content clippings
+        history.push(`/content/${props.modelZUID}/${items[1].meta.ZUID}`);
       }
     }
+  }, [props.modelZUID]);
+
+  // on initial and
+  // on modelZUID change
+  // update filter and load fields/items
+  useEffect(() => {
+    setFilter(
+      props.filters[props.modelZUID]
+        ? props.filters[props.modelZUID]
+        : DEFAULT_FILTER
+    );
+    load(props.modelZUID);
+  }, [props.modelZUID]);
+
+  // on filter change
+  // run and persist filters
+  useEffect(() => {
+    if (_isMounted.current) {
+      runFilters();
+      persistFilters();
+    }
+  }, [filter]);
 
-    componentWillUnmount() {
-      this._isMounted = false;
-      // save filter data to store
-      this.props.dispatch({
-        type: "PERSIST_LIST_FILTERS",
-        modelZUID: this.props.modelZUID,
-        filters: {
-          filterTerm: this.state.filterTerm,
-          sortedBy: this.state.sortedBy,
-          status: this.state.status,
-          colType: this.state.colType,
-          reverseSort: this.state.reverseSort
-        }
-      });
+  // HACK: runFilters after waiting for new props
```

Comment:
The majority of places where you have included a `// HACK` comment need more information on why it is a hack. 
e.g. > runFilters after waiting for new props

Is this a hack vs running the filters synchronously? The "hack" on the `useEffect` is confusing because what is described is the default behavior of `useEffect`. i.e. waiting for new props.